engine/schema/src/main/resources/META-INF/db/schema-41720to41800.sql

```diff
@@ -19,6 +19,10 @@
 -- Schema upgrade from 4.17.2.0 to 4.18.0.0
 --;
 
+-- Add support for VMware 8.0
+INSERT IGNORE INTO `cloud`.`hypervisor_capabilities` (uuid, hypervisor_type, hypervisor_version, max_guests_limit, security_group_enabled, max_data_volumes_limit, max_hosts_per_cluster, storage_motion_supported, vm_snapshot_enabled) values (UUID(), 'VMware', '8.0', 1024, 0, 59, 64, 1, 1);
+INSERT IGNORE INTO `cloud`.`guest_os_hypervisor` (uuid,hypervisor_type, hypervisor_version, guest_os_name, guest_os_id, created, is_user_defined) SELECT UUID(),'VMware', '8.0', guest_os_name, guest_os_id, utc_timestamp(), 0  FROM `cloud`.`guest_os_hypervisor` WHERE hypervisor_type='VMware' AND hypervisor_version='7.0.3.0';
+
```

Comment:
@rohityadavcloud 
the upgrade of SDK to 8.0 looks good.

Anyone has tested the hypervisor and guest os mappings of vmware 8 ?
It not, it might be better to add it in a separated PR.

Reply:
The upgrade of SDK can be merged now

Reply:
@rohityadavcloud ^^?

Reply:
Sorry I didn't get your comments earlier @weizhouapache @DaanHoogland

I'm of the opinion that SDK upgrade isn't just enough you need to add (1) the new hypervisor in hypervisor capability, and (2) copy the guest OS mappings from the previous hypervisor version. You an review the previous PRs that added support for vcenter 7.0 (and it's minor versions), 6.7, 6.5 etc. they all follow the same pattern.

In the SDK, there's a utility and enums that translate the guest OS label to what it means in cloudstack DB (and vice-versa), because of this you can copy the guest OS mappings in CloudStack DB. This list is here too: https://docs.vmware.com/en/VMware-HCX/4.5/hcx-user-guide/GUID-D4FFCBD6-9FEC-44E5-9E26-1BD0A2A81389.html

So in my opinion, as long as regression tests are passing with the new SDK it's fair to say we can copy the mappings to the guest_os_hypervisor table. We may later still need to do these things (a) add support for the template in trillian/BO, (b) review and add new guest OSs that the new vSphere version supports incl. fixing any incorrect mapping (if any), (c) ensure smoketests pass with the new template (from trillian/BO). These addtional things can be done in 4.18.1.0.

Reply:
Hi @rohityadavcloud 
in my opinion, the SDK upgrade and vmware 8 support are two different tasks.
I am fine with SDk upgrade, but hesitated to merge the SQL changes for vmware 8 (code looks good but it is not tested manually or automatedly).

Reply:
I can try to spin a new env locally.